packages/generator/src/complex-type/field-type-class.ts

```diff
@@ -17,10 +17,28 @@ export function fieldTypeClass(
   return {
     kind: StructureKind.Class,
     name: `${complexType.fieldType}<EntityT extends Entity>`,
-    extends: 'ComplexTypeField<EntityT>',
+    extends: `ComplexTypeField<EntityT, typeof ${complexType.typeName}>`,
     isExported: true,
     properties: properties(complexType),
-    docs: [getComplexTypeFieldDescription(complexType)]
+    docs: [getComplexTypeFieldDescription(complexType)],
+    ctors: [
```

Comment:
Is this short for constructors? I would prefer a more speaking name or is this a common abbreviation?

Reply:
This is the ts-morph API